### ISSUE TITLE
Select window of popup

### ISSUE THREAD

**Issue by aagaard** [bmag/emacs-purpose#69]
It is possible for purpose to select certain windows when they popup?

**Comment by bmag** [comment]
*TL;DR: use the hook `purpose-display-buffer-functions`*

Usually yes, but it depends on how the window is popped up in the first place (implementation details). After Purpose displays a buffer, it runs the hook `purpose-display-buffer-functions` - you can use it to select the windows you want when they popup.

Note that if the original code doesn't rely on `switch-to-buffer` or `display-buffer`, but instead uses `set-window-buffer` directly, then Purpose won't be able to help you. Usually `display-buffer` is sufficient, but I've seen a few packages that use `set-window-buffer` directly.

Note also that there are several different commands for displaying a buffer, each with slightly different semantics. Those are the `switch-to-buffer(-*)`, `pop-to-buffer(-*)` and `display-buffer` commands. If the original code relies on the displayed buffer not being selected, you might see a buggy behavior.

Could you tell me which windows specifically you have in mind?

**Comment by aagaard** [comment]
Thank you for your quick response. Yes, for instance, all help-mode, grep-mode and info-mode buffers.

**Comment by bmag** [comment]
Well, I think you shouldn't have trouble with these. Please tell me if you're having trouble, or close this issue if you succeeded.

**Comment by bmag** [comment]
@aagaard any news on this?
